src/main/groovy/life/qbic/portal/sampletracking/DependencyManager.groovy

```diff
@@ -109,14 +114,16 @@ class DependencyManager {
         }
 
         try {
+            final def presenter = new SampleListPresenter(viewModel)
             SampleTrackingUpdateDataSource trackingUpdateCenter = SampleTracker.createSampleTrackingUpdate(trackingServices.get(0), this.serviceUser)
-            this.updateInfoInteractor = new UpdateSampleTrackingInfo(trackingUpdateCenter, this.viewModel as SampleTrackingUpdateOutput)
+            this.updateInfoInteractor = new UpdateSampleTrackingInfo(trackingUpdateCenter, presenter)
         } catch (Exception e) {
             log.error("Could not setup ${UpdateSampleTrackingInfo.getSimpleName()} use case", e)
         }
 
         try {
-            this.modifySampleListInteractor = new ModifySampleList(this.viewModel as ModifySampleListOutput)
+            final def presenter = new SampleImportPresenter(viewModel)
+            this.modifySampleListInteractor = new ModifySampleList(presenter)
```

Comment:
```suggestion
            final def sampleImportPresenter = new SampleImportPresenter(viewModel)
            this.modifySampleListInteractor = new ModifySampleList(sampleImportPresenter)
```
I get that this does not leave the `try` block but it might show that the `try` blocks create different presenters.